joblib/_parallel_backends.py

```diff
@@ -565,10 +568,27 @@ def wrap_future_result(future, timeout=None):
 
     def terminate(self):
         if self._workers is not None:
+            # Trigger the collection of temporary shared memory created to by
+            # joblib by signaling the resource_tracker we don't need it
+            # anymore.
+            for filename in JOBLIB_MMAPS:
+                resource_tracker.maybe_unlink(filename, "file")
+            JOBLIB_MMAPS.clear()
```

Comment:
The call to terminate on a particular `joblib.Parallel` instance should not have side effect on other `joblib.Parallel` call running in other Python threads of the same process. This means that the list of tempory memmap filenames should be stored as an attribute of the Parallel instance instead of a global variable / module attribute.

So instead we should use a new attribute, e.g.: `self.parallel._tempory_memmap_filenames`.

This also mean that this per-Parallel instance attribute should be passed as an argument to the  `ArrayMemmapReducer` instance used for forward reducing.

Reply:
Actually scratch that: we can just list the filenames of the `self._workers._temp_folder` directly. There is no need for using a `JOBLIB_MMAPS` at all here.

Reply:
You're right, it's a better alternative.